### PR TITLE
add function: getPagesInNamespace

### PR DESCRIPTION
**Pull request by kyv**
Add a function to easily pull the pages from a given namespace. Example of use: 

    client.getPagesInNamespace(502, function(err, pages) {
        console.log('All pages: %d', pages.length);
        console.log(JSON.stringify(pages.slice(0, 50)));

        // get all revisions of a single article
        var pageId = parseInt(pages[0].pageid, 10);

        client.getArticleRevisions(pageId, function(err, revisions) {
            console.log(JSON.stringify(revisions));
        });
    });

### CODE REVIEW

File: lib/bot.js [hunk 1]
```diff
@@ -264,6 +264,20 @@ module.exports = (function() {
 			);
 		},
 
+        getPagesInNamespace: function(namespace, callback) {
+            this.log("Getting pages in namespace " + namespace);
+
+            this.api.call({
```
Comment (macbre) [inline lib/bot.js]: Please use `this.getAll` instead (see `getPagesInCategory` above) to iterate over all pages.

File: lib/bot.js [hunk 2]
```diff
@@ -264,6 +264,20 @@ module.exports = (function() {
...
+                action: 'query',
+                list: 'allpages',
+                apnamespace: namespace,
+                apfilterredir: 'nonredirects', // do not include redirects
+                aplimit: 5000
+            }, function(err, data) {
+                callback(err, data && data.allpages || []);
+            });
+        },
+
```
Comment (macbre) [inline lib/bot.js]: Tabs instead of spaces, please

File: lib/bot.js [hunk 3]
```diff
@@ -264,6 +264,20 @@ module.exports = (function() {
 			);
 		},
 
+        getPagesInNamespace: function(namespace, callback) {
```
Comment (macbre) [inline lib/bot.js]: Please document this function in the `README.md` file

**Comment by macbre** [comment]
Thanks for the PR! Inline comments left.

**Comment by kyv** [comment]
Please see 8e1abf9 and b4debfb

**Comment by macbre** [comment]
Please update the README and we're ready to merge

**Comment by kyv** [comment]
4f44595 Adds the function to the README. I linked to http://www.mediawiki.org/wiki/API:Allpages. I hope that is correct.

**Comment by macbre** [comment]
@kyv, thanks!

**Pull request merged**